src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -77,10 +83,14 @@ private async Task<string> GetRegionAsync(ICoreLogger logger)
                 logger.Info($"[Region discovery] Call to local IMDS returned region: {localImdsResponse.location}");
                 return localImdsResponse.location;
             }
+            catch (MsalClientException)
+            {
+                throw;
+            }
```

Comment:
why ?

Reply:
Because at line 76 I am checking the status of the request and if it is not a 200 then throwing a proper exception at line 76. But in case there is failure to make a call to local imds or any other exception occurs, I am catching it at line 90 and forming the above exception.